### ISSUE TITLE
Font has excessive gaps between characters

### ISSUE THREAD

**Issue by ryanpeach** [hlissner/doom-emacs#3692]
![image](https://user-images.githubusercontent.com/510883/89447462-fd329800-d723-11ea-89da-2be315aa9f38.png)

![image](https://user-images.githubusercontent.com/510883/89447472-01f74c00-d724-11ea-899b-3bb738b84cd3.png)

![image](https://user-images.githubusercontent.com/510883/89447487-058ad300-d724-11ea-9ea8-bbf56f9dd8b3.png)

<details>
<summary>Additional details</summary>

```elisp
;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-

;; Place your private configuration here! Remember, you do not need to run 'doom
;; sync' after modifying this file!

;; Personal default keybindings
(map!
 :leader
 "SPC" #'counsel-M-x
 "p l" #'projectile-switch-project
 "w w" #'ace-window
 "w n" #'evil-window-next
 "w p" #'evil-window-previous
 "g s" #'magit-stage-file
 "g S" #'magit-stage-modified
 "g p" #'magit-pull
 "g P" #'magit-push
 "p /" #'+ivy/project-search
 "TAB TAB" #'indent-rigidly
 )

;; Handle window splitting and size
;; REF: https://www.reddit.com/r/emacs/comments/a7wsp6/resize_frame_when_creating_new_windows/
; Set default frame size
(setq default-window-width 99) ; space included for fci-mode rule, ultimately comes to 79 characters, the PEP8 default
(when window-system (set-frame-size (selected-frame) default-window-width 48))

; Resize frame when opening new windows hor1izontally
(defun delete-other-windows-and-resize ()
  "delete and reset width"
  (interactive)
  (when window-system
    (progn (set-frame-width (selected-frame) default-window-width)
           (delete-other-windows))))

(defun delete-window-and-resize ()
  "delete this window and reset width"
  (interactive)
  (when window-system
    (progn (delete-window)
           (set-frame-width (selected-frame)
                            (- (frame-width) default-window-width 1))
           (balance-windows))))

(defun split-window-right-and-resize ()
  "split and increase width"
  (interactive)
  (when window-system
    (progn (set-frame-width (selected-frame)
			     (+ (frame-width) default-window-width 1))
	    (split-window-right)
	    (balance-windows))))

(map!
 :leader
 "w d" #'delete-window-and-resize
 "w f" #'delete-other-windows-and-resize
 "w \"" #'split-window-right-and-resize
 "w %" #'split-window-vertically)

;; ivy edit mode made easier
;; TODO: Fix this, it hangs for some reason
(defun ivy-edit ()
  "Runs ivy-occur then ivy wgrep at the same time"
  (interactive)
  (progn (ivy-occur)
         (ivy-wgrep-change-to-wgrep-mode)))

(map!
 :leader
 "/ /" #'+ivy/project-search
 )

(after! ivy
  (define-key ivy-minibuffer-map (kbd "C-c e") 'ivy-edit)
  (define-key ivy-minibuffer-map (kbd "C-c o") 'ivy-occur)
  (define-key ivy-minibuffer-map (kbd "C-c q") 'ivy-wgrep-change-to-wgrep-mode)
)

;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Ryan Peach"
      user-mail-address "ryan.peach@outlook.com")

;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:
(setq doom-font (font-spec :family "monospace" :size 14 :average-width 8 :space-width 8 :max-width 8))

;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
(setq doom-theme 'doom-one)

;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/org/")

;; This determines the style of line numbers in effect. If set to `nil', line
;; numbers are disabled. For relative line numbers, set this to `relative'.
(setq display-line-numbers-type "relative")


;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c g k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c g d') to jump to their definition and see how
;; they are implemented.
```

</details>

I tried this
```
(setq doom-font (font-spec :family "monospace" :size 14 :average-width 8 :space-width 8 :max-width 8))
```
and it didn't do anything.

My `describe-font` screens are in the imgur.

Surprisingly, if there is a bug in this code (like i put in a syntax error), doom goes back to default and everything is fine text wise.

**Comment by hlissner** [comment]
This is a common issue. `monospace` is not the name of a font. It's the name of a family of fonts, and by selecting it, your OS will use the first valid font in that family. What font does it select? That's entirely up to your local fontconfig settings and what fonts you have installed. 

The problem is that whatever font it's choosing is either broken, corrupt, incompatible, or not actually a monospaced font (i.e. fontconfig couldn't find a valid fonts in the monospace family, so it fell back to another unknown family).

**TL;DR:** Either:

1. Change your `doom-font` to an specific font,

   ```elisp
   (setq doom-font (font-spec :family "Fira Code" :size 14))
   ```

2. Remove the `(setq doom-font ...)` statement entirely and let your system choose a font for you.

3. Fix your fontconfig configuration. There's a decent line up of common issues and example configuration in [this entry of the Arch wiki](https://wiki.archlinux.org/index.php/Font_configuration).

As this is not a Doom issue I will close this, but feel free to follow up here.
